prog/vnet/load_balancer_nexus.rb

```diff
@@ -43,11 +46,72 @@ def before_run
       decr_rewrite_dns_records
     end
 
+    when_refresh_cert_set? do
+      hop_refresh_cert
+    end
+
     hop_create_new_health_probe if strand.children_dataset.count < load_balancer.vms_dataset.count
 
     nap 5
   end
 
+  label def refresh_cert
+    private_key = OpenSSL::PKey::RSA.new(4096)
+    client = Acme::Client.new(private_key: private_key, directory: Config.lets_encrypt_directory)
+    account = client.new_account(contact: "mailto:#{Config.lets_encrypt_email}", terms_of_service_agreed: true)
+    order = client.new_order(identifiers: [load_balancer.hostname])
+    authorization = order.authorizations.first
+    load_balancer.update(kid: account.kid, private_key: private_key, order_url: order.url)
+    dns_challenge = authorization.dns
+    Prog::Vnet::LoadBalancerNexus.dns_zone.insert_record(record_name: "#{dns_challenge.record_name}.#{load_balancer.hostname}", type: dns_challenge.record_type, ttl: 600, data: dns_challenge.record_content)
```

Comment:
by the way, just ot be particular, I noticed you truncated the ubids for load balancer's entropic hostnames, do you have unique constraints to prevent Undefined Stuff from happening if a collision in the first N bits is generated? (we should probably do this for VM-to-Host assignment as well, but it kinda slid in from the prototype era)

Reply:
I have replaced it with private subnet's ubid. This way, we won't need to generate a specific cert for every load balancer in the same subnet. They can share. I was planning to start generating wildcard certs like `*.[private_subnet_ubid_last_5_chars].lb.ubicloud.com`